adoc/chapters/programming_interface.adoc

```diff
@@ -5190,708 +5190,597 @@ include::{header_dir}/buffer.h[lines=4..-1]
-      [code]#buffer# can be shared with the application via a
-      [code]#std::mutex# provided to the property. The mutex
-      [code]#m# is locked by the runtime whenever the data is in
-      use and unlocked otherwise. The contents of [code]#hostData# are
-      guaranteed to reflect the contents of the buffer when the
-      [code]#std::mutex# is unlocked by the runtime.
+_Effects (1):_ Constructs an [code]#use_host_ptr# property object.
 
-a@
-[source]
-----
-property::buffer::context_bound
-----
-   a@ The [code]#context_bound# property adds the requirement that the SYCL [code]#buffer# can only be associated with a single SYCL [code]#context# that is provided to the property.
+'''
 
-|====
+.[apidef]#property::buffer::use_mutex#
+[source,role=synopsis,id=api:property-buffer-use-mutex]
+----
+namespace sycl::property::buffer {
+class use_mutex {
+  use_mutex(std::mutex& mutexRef);
 
+  std::mutex* get_mutex_ptr() const;
+};
+} // namespace sycl::property::buffer
+----
 
-The constructors and special member functions of the buffer property classes are
-listed in <<table.constructors.properties.buffer>> and
-<<table.members.properties.buffer>> respectively.
+The [code]#use_mutex# property is valid for the SYCL [code]#buffer#,
+[code]#unsampled_image# and [code]#sampled_image# classes.
+The property adds the requirement that the memory which is owned by the SYCL
+[code]#buffer# can be shared with the application via a [code]#std::mutex#
+provided to the property.
+The mutex [code]#m# is locked by the runtime whenever the data is in use and
+unlocked otherwise.
+The contents of [code]#hostData# are guaranteed to reflect the contents of the
+buffer when the [code]#std::mutex# is unlocked by the runtime.
 
+'''
 
-[[table.constructors.properties.buffer]]
-.Constructors of the [code]#buffer# [code]#property# classes
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Constructor @ Description
-a@
-[source]
+.[apidef]#property::buffer::use_mutex constructor#
+[source,role=synopsis,id=api:property-buffer-use-mutex-ctor]
 ----
-property::buffer::use_host_ptr::use_host_ptr()
+use_mutex(std::mutex& mutexRef);
 ----
-   a@ Constructs a SYCL [code]#use_host_ptr# property instance.
 
-a@
-[source]
+_Effects:_ Constructs a SYCL [code]#use_mutex# property instance with a
+reference to [code]#mutexRef# parameter provided.
+
+'''
+
+.[apidef]#property::buffer::use_mutex::get_mutex_ptr#
+[source,role=synopsis,id=api:property-buffer-use-mutex-get-mutex-ptr]
 ----
-property::buffer::use_mutex::use_mutex(std::mutex& mutexRef)
+std::mutex* get_mutex_ptr() const;
 ----
-   a@ Constructs a SYCL [code]#use_mutex# property instance with a reference to [code]#mutexRef# parameter provided.
 
-a@
-[source]
-----
-property::buffer::context_bound::context_bound(context boundContext)
+_Returns:_ The [code]#std::mutex# which was specified when constructing this
+SYCL [code]#use_mutex# property.
```

Comment:
```suggestion
_Returns:_ A pointer to the [code]#std::mutex# provided when constructing this property.
```

Kind of a nit, but I like when we can avoid being verbose.  I considered "provided at construction" too, but I don't know if that's clear enough.